src/web-ui/src/App.js

```diff
@@ -68,119 +58,119 @@ class App extends Component {
         gateway.postEngagement(result).then(() => {});
         return result;
       });
-
-      this.setState({ detectedFaces }, () => {
-        if (this.state.rekognizing) {
-          setTimeout(this.getSnapshot, 300);
-        }
-      });
+      setDetectedFaces(detectedFaces);
+      setIterationCompleted(true);
     });
 
     gateway.searchFaces(b64Encoded).then(response => {
       const detectedPeople = [];
```

Comment:
Not related to this PR but this code could be simplified, I think:

```js
const detectedPeople = response.FaceMatches.filter(match => {
      const externalImageId = match.Face.ExternalImageId;
      return Boolean(people.find( x => x.externalImageId === externalImageId))
    });
```

Reply:
Be careful with that, I was thinking the exact same thing yesterday night but the return value is different, you might need a map later on

Reply:
I'm not sure I understand, why is the return type different?

Reply:
because with a filter you are returning the whole object, whatever is `FaceMatches` (I don't have clue), look at what instead is `detectedPeople` array? it's a `people` (whatever people is) so looking at this code, two completely different objects right?

Reply:
Ah yes, of course! Excuse, me I'm still half asleep this morning.

Reply:
I'm answering because I thought the exact same thing, and then I bite my tong and delete the comment :-P

Reply:
Guess it's down to good old fashioned `reduce` then:

```js
const detectedPeople = FaceMatches.reduce((acc, curr) => {
  const externalImageId = curr.Face.ExternalImageId;
  const matched = people.find(x => x.externalImageId === externalImageId);
  return matched ? (acc.push(matched), acc) : acc
}, []);
```

Reply:
or 
```
const detectedPeople = FaceMatches.map(match => people.find(x => x.externalImageId === match.Face.ExternalImageId)).filter(Boolean)
```
if you want, but why bother?

Reply:
Because it's fun! 😄

Reply:
Btw, I actually think yours is the clearest out of all three versions.

Reply:
Clearest but probably the worst performance between the three.
Front end people don't usually care too much at performance even because we are talking about probably nothing in case of list of dozen or even hundred of elements.
Back end people likes loops, and probably forEach or reduce is not good enough, you can't exit the loop omg :P.
Anyway your choice, I prefer a map and filter usually

Reply:
I actually reverted this. Turns out, I was checking if `find` would match a value because I was storing `people` into the state, and on the first render that was not matching the results in `detectedPeople` because of the immutability of the function. By storing people as ref, the problem on the first render doesn't appear and I actually can get updated results, as it was with the class originally. So, `useRef` to the rescue again 💃